src/server/cluster/outgoing_slot_migration.cc

```diff
@@ -31,12 +31,14 @@ namespace dfly::cluster {
 class OutgoingMigration::SliceSlotMigration : private ProtocolClient {
  public:
   SliceSlotMigration(DbSlice* slice, ServerContext server_context, SlotSet slots,
-                     journal::Journal* journal)
+                     journal::Journal* journal, OutgoingMigration* om)
       : ProtocolClient(server_context), streamer_(slice, std::move(slots), journal, &cntx_) {
+    cntx_.SwitchErrorHandler([om](auto ge) { om->Finish(std::move(ge)); });
```

Comment:
what cluster test do we have that checks the logic that you introduced ? do we have an reconnect during migration pytest?

Reply:
test_network_disconnect_during_migration

Reply:
we need to change this test a little bit
1. use static seeder and increase the key num to 300K
2. add wait for migration finish before push finilize config
3. add a step to make sure you wait for snapshoting to start before you drop proxy